### ISSUE TITLE
References sometimes not syncing for GSDs

### ISSUE THREAD

**Issue by gdower** [Sp2000/colplus-backend#578]
For some GSDs the references are totally missing in the ac-export. For example these are the only references included in references.csv of [export-2053.zip](http://api.catalogue.life/download/export-2053.zip) for FishBase:

```
grep ',10,' references.csv
102308,\N,\N,\N,\N,10,ab0b7d5c-fc00-4bd8-b8d4-d96e45ebb297
112370,\N,\N,\N,\N,10,ddd18db1-2d19-4f65-855f-2b340abd2402
112404,\N,\N,\N,\N,10,5659c8be-fc11-4ab3-909b-d2a732b248d3
112405,\N,\N,\N,\N,10,7c730a16-0ac7-490c-a90d-2e80420e5b1f
112415,\N,\N,\N,\N,10,5fc4cc59-4626-4b42-b6b4-e74cc4c53ff1
112424,\N,\N,\N,\N,10,e07aebeb-0544-4ecd-93c7-a06a271f8eb7
112432,\N,\N,\N,\N,10,2c18d9a9-6bb4-4ee2-91a0-4adddbba1189
112440,\N,\N,\N,\N,10,5d9aefb3-31ae-4990-b5b8-77ccf61a736f
112441,\N,\N,\N,\N,10,14fd3835-351a-443c-b343-eed8e85dd5d9
112442,\N,\N,\N,\N,10,b8b7fdd7-5951-4882-9235-9a8174f9623e
112443,\N,\N,\N,\N,10,13cecc6e-06d0-4493-8d71-b49d651f0111
112444,\N,\N,\N,\N,10,af09b961-7572-4493-90d3-18f094c2c8a5
112445,\N,\N,\N,\N,10,6813b0ce-69d8-47c2-8e02-dab8f9b06981
112446,\N,\N,\N,\N,10,ec2a0a69-8110-4375-ad0e-92953972183b
112565,\N,\N,\N,\N,10,f94c8499-ecd5-47c4-8f03-85dcc37c45a7
112566,\N,\N,\N,\N,10,7e77e9c5-d3e3-4311-ab86-2fab8c134f22
112567,\N,\N,\N,\N,10,2aea17ce-2942-4986-89c9-7a02da99e07c
112568,\N,\N,\N,\N,10,f3be8c54-ae13-40a1-8e01-51bb1b85c21e
112569,\N,\N,\N,\N,10,2ec06493-4fac-4865-a980-fc3d899fad46
112570,\N,\N,\N,\N,10,6cc5af94-af98-40d0-aafb-3730895e30d3
```

I'm going to re-run the col-diff-tool on a random sample of scientific names catalogue-wide to try to find out more info on this bug. Originally, I thought this was caused by a 4 character truncation of the CoL+ IDs (because of the 4 -'s added to the CoL+ IDs) by ConversionTool, but there's obviously some other ac-exporter issue since there's so few references included and the [FishBase references are in the Clearinghouse](https://data.catalogue.life/dataset/1010/reference?limit=50&offset=0).

Ref: https://github.com/Sp2000/colplus-repo/issues/89

**Comment by gdower** [comment]
@mdoering, I think this is actually a sector importer problem, because the references won't load in /dataset/3 and are indeed [missing](https://data.catalogue.life/dataset/3/reference?limit=50&offset=0&q=Check-list%20of%20the%20fishes%20of%20the%20eastern%20tropical%20Atlantic) compared with [dataset/1010](https://data.catalogue.life/dataset/1010/reference?limit=50&offset=0&q=Check-list%20of%20the%20fishes%20of%20the%20eastern%20tropical%20Atlantic):

![image](https://user-images.githubusercontent.com/8573609/70179939-99dbfc00-16a4-11ea-9bad-13ce528f8e9e.png)

**Comment by gdower** [comment]
Here's the [logs](https://logs.gbif.org/app/kibana#/discover?_g=(refreshInterval:(display:On,pause:!f,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:AWyLa4mQHCKcR6PFXu4X,key:sector,negate:!f,type:phrase,value:3012),query:(match:(sector:(query:212,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:AWyLa4mQHCKcR6PFXu4X,key:environment,negate:!f,type:phrase,value:col-prod),query:(match:(environment:(query:col-prod,type:phrase))))),index:AWyLa4mQHCKcR6PFXu4X,interval:auto,query:(match_all:()),sort:!('@timestamp',desc))) for the FishBase Actinopterygii sector sync.

**Comment by mdoering** [comment]
indeed. Missing for all vernaculars and for the synonyms the keys are there, but the actual references seem missing

**Comment by mdoering** [comment]
fixed. After syncing the `Elasmobranchii` sector of Fishbase:

<img width="1111" alt="Bildschirmfoto 2019-12-06 um 17 03 25" src="https://user-images.githubusercontent.com/327505/70337026-984c3880-184a-11ea-898b-4cc1dc53adf6.png">
<img width="1116" alt="Bildschirmfoto 2019-12-06 um 17 04 56" src="https://user-images.githubusercontent.com/327505/70337027-984c3880-184a-11ea-8f0d-c6af271e06cd.png">

**Comment by gdower** [comment]
@mdoering, I'm reopening, because I think there's still some issue that is causing full sectors or partial sectors to lose references. I'll try to figure out why.

Here's the output from 25 random draws from each GSD:

http://ower.org/col_diff/2019-11-25_random_by_gsd/

There are some GSDs were references totally failed to sync into assembly and others where only some species are missing references. Here's an example from FLOW, which has only 1 sector in CoL, so it looks like sometimes references are sporadically failing to sync for some but not all species in the sector:

[Source (Ref present)](https://data.catalogue.life/dataset/1011/taxon/13433) vs [Assembly (Ref missing)](https://data.catalogue.life/dataset/3/taxon/32f2e55b-ab3b-4eb5-a2ce-0c57fec6bf7f)
![image](https://user-images.githubusercontent.com/8573609/70336124-379f0a80-180e-11ea-8f44-de2027f19c18.png)

**Comment by gdower** [comment]
@mdoering, I didn't see this on gitter until now:

> I think I also fixed the reference sync bug
> was again datasetKey vs subjectDatasetKey :)

Should I do a full resync on prod to see if that fixes the other missing references?

**Comment by gdower** [comment]
Re-syncing the FLOW sector didn't work: https://data.catalogue.life/dataset/3/taxon/95d5b429-5649-43d1-8051-b6ceeceefcbb

**Comment by mdoering** [comment]
we have not deployed to prod. its all dev for now

**Comment by mdoering** [comment]
@gdower @thomasstjerne @ayco-at-naturalis I have created a new label `deployed` that we should tag a closed issue with after the code was actually deployed to prod. Issues should be closed when the code is patched in the master branch and it is useful to close an issue via a commit line using fixes #1234. This links the actual code to the issue. But it does not mean it is deployed to prod yet so not necessarily visible for users/testers. The new label hopefully helps to clarify that.
